src/lib/sys/abstract_configuration.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <inttypes.h>
+#include <string.h>
+
+namespace cfg {
+
+// Allows for mocking in unit tests
+class AbstractConfiguration {
+public:
+  virtual bool getConfigCbor(const char *key, size_t key_len, uint8_t *value,
+                             size_t &value_len) = 0;
+  virtual bool setConfigCbor(const char *key, size_t key_len, uint8_t *value,
+                             size_t value_len) = 0;
+};
```

Comment:
Obviously the real configuration class has many more methods. These two were all I needed for the test mocks though. If in the future we write more tests using the MockConfiguration and need more of the methods mocked, then they just need to be added both in the mock class and here.